inc/commonitilobject.class.php

```diff
@@ -6263,66 +6222,126 @@ static function showShort($id, $options = []) {
+                                                ($p['output_type'] == Search::HTML_OUTPUT?'<br>':''),
+                                             getUserName($plan['users_id_tech']));
+                  }
+                  $planned_infos .= "<br>";
                }
-               $planned_infos .= "<br>";
+
             }
 
-         }
+            $tenth_column = count($items);
+            if ($tenth_column) {
+               $tenth_column = "<span class='pointer'
+                                 id='".$item->getType().$item->fields["id"]."planning$rand'>".
+                                 $tenth_column.'</span>';
+               $tenth_column = sprintf(__('%1$s %2$s'), $tenth_column,
+                                       Html::showToolTip($planned_infos,
+                                                         ['display' => false,
+                                                               'applyto' => $item->getType().
+                                                                              $item->fields["id"].
+                                                                              "planning".$rand]));
+            }
 
-         $tenth_column = count($items);
-         if ($tenth_column) {
-            $tenth_column = "<span class='pointer'
-                              id='".$item->getType().$item->fields["id"]."planning$rand'>".
-                              $tenth_column.'</span>';
-            $tenth_column = sprintf(__('%1$s %2$s'), $tenth_column,
-                                    Html::showToolTip($planned_infos,
-                                                      ['display' => false,
-                                                            'applyto' => $item->getType().
-                                                                           $item->fields["id"].
-                                                                           "planning".$rand]));
+            echo Search::showItem($p['output_type'], $tenth_column, $item_num, $p['row_num'],
+                                 $align_desc." width='150'");
+         } else {
+            echo Search::showItem($p['output_type'], $name_column, $item_num, $p['row_num'], $align_desc." width='200'");
+            $takeintoaccountdelay_column = Html::timestampToString($item->fields['takeintoaccount_delay_stat']);
+            echo Search::showItem($p['output_type'], $takeintoaccountdelay_column, $item_num, $p['row_num'], $align_desc." width='150'");
+
+            $solvedelay_column = "";
+            // Show only for solved tickets
+            if ($item->fields['solve_delay_stat'] > 0) {
+               $solvedelay_column = Html::timestampToString($item->fields['solve_delay_stat']);
+            }
+            echo Search::showItem($p['output_type'], $solvedelay_column, $item_num, $p['row_num'], $align_desc." width='150'");
+
+            $waiting_duration_column = Html::timestampToString($item->fields['waiting_duration']);
```

Comment:
It may happen that this value is not yet set, it should be handled.

Reply:
I think this one is different that the two above since it is not a distinct "milestone" in the ticket lifecycle but a representation of the ticket spent in one specific state.

It would be bad to display the `solve_delay_stat` value before the ticket was solved because it would read as '0 seconds happened before the ticket was solved' which is not true.

However for `waiting_duration`  it read as 'This ticket has been pending for 0 seconds' which is true.